### PR TITLE
2234 scores per level

### LINKED ISSUE #2234: Redesign scores per level assignments viz in D3
Probably as a histogram showing frequency of scores earned

### PR DESCRIPTION
**Pull request by mariehooper**
### Status
**IN DEVELOPMENT**

### Description
This PR updates the 'Scores Earned' chart on the Class Analytics page for each graded assignment for students and instructors as well as the Learners Participated donut chart.

### Impacted Areas in Application
List general components of the application that this PR will affect:

* Instructor and student class analytics for assignments

======================
Closes #2234 #2233 #3062

### CODE REVIEW

**Comment by mariehooper** [comment]
@chcholman - When you want to take a look at this, please do! The sample data we have is a lot of very staggered point assignment where there are often unique scores for almost all the students in the course. I updated one of the assignments myself with some semi-realistic data and got something that looks like this:

screen shot 2017-03-03 at 8 59 29 am

Would love to see this against staging data like we discussed, as I'm sure I'll need to make some considerations for label positioning on the x-axis, etc. And pay less attention to the arrangement of graphs on the analytics page as I plan to shift these around when the final graph is updated. :)

**Comment by chcholman** [comment]
@mariehooper Awesome! I'll get this into staging next. Is there any way to control the Y axis on the Scores Earned bar chart? Since it will always be based on count, there's no need for it to go into decimals (it's doing this locally).

**Comment by chcholman** [comment]
@mariehooper This is looking so great! Still to do:

- [] Labels on X & Y axis for Scores earned 
- [] Make Y axis whole numbers only (no half people!)
- [] When the boxplot is hidden, the participant count is displaying oddly: 
screen shot 2017-03-06 at 4 27 15 pm

**Comment by mariehooper** [comment]
@chcholman Backatcha! :)

File: app/assets/javascripts/assignment-results-vis.js [hunk 1]
```diff
@@ -53,5 +53,48 @@ if ($assignmentDistro.length) {
...
+      b: 40,
+      t: 4,
+      pad: 8
+    },
+    annotations: [{
+      font: {
+        size: 16
+      },
+      showarrow: false,
+      text: 'Learners<br> Participated',
```
Comment (chcholman) [inline app/assets/javascripts/assignment-results-vis.js]: Eventually, this needs to be responsive to the term for students

File: app/assets/javascripts/levels_per_assign.js [hunk 2]
```diff
@@ -1,54 +1,88 @@
...
+      t: 4,
+      pad: 8
+    },
+    xaxis: {
+      fixedrange: true,
+      title: 'Score'
+    },
+    yaxis: {
+      fixedrange: true,
+      title: '# of Students',
```
Comment (chcholman) [inline app/assets/javascripts/levels_per_assign.js]: (same as above)

**Pull request merged**